### PR TITLE
Add layer manager component

### LINKED ISSUE #1337: Composant gestion des couches
Componsant simple permetant de rendre les couches visibles/invisibles (avec des cases à cocher).

### PR DESCRIPTION
**Pull request by pgiraud**
Fixes #1337

### CODE REVIEW

**Comment by pgiraud** [comment]
Ready for review.

File: jsapi/externs/luxx.js [hunk 1]
```diff
@@ -150,3 +157,16 @@ luxx.MousePositionOptions.prototype.srs;
...
+ * @typedef {{ target: (Element|string) }}
+ */
+luxx.ControlOptions;
+
+/**
+ * The container for the control UI component, either the element itself or
+ * the `id` of the element.
+ * @type {Element|string}
+ */
+luxx.ControlOptions.prototype.target;
```
Comment (tonio) [inline jsapi/externs/luxx.js]: There are others control with `target` as single property for the moment. I wonder what’s better between a generic one (I don’t think you can extend typedef) or, specific one to be easily extended.

I generic one is chosen (`ControlOptions`), at least `BgSelectorOptions` should be updated to use it.

File: jsapi/src/main.js [hunk 2]
```diff
@@ -134,6 +135,13 @@ lux.Map = function(options) {
   goog.base(this, options);
 
   this.getTargetElement().classList.add('lux-map');
+
+  if (options.layerManager) {
+    var control = new lux.LayerManager({
+      target: options.layerManager.target
```
Comment (tonio) [inline jsapi/src/main.js]: Missing check for element existence as for other controls.

File: jsapi/src/main.js [hunk 3]
```diff
@@ -318,6 +326,7 @@ lux.WMTSLayerFactory_ = function(config) {
       style: 'default'
     })
   });
+  layer.set('name', config.name);
```
Comment (tonio) [inline jsapi/src/main.js]: [Already set](https://github.com/Geoportail-Luxembourg/geoportailv3/blob/jsapi/jsapi/src/main.js#L350)?

File: jsapi/src/layermanager.js [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+    var li = document.createElement('li');
+    var input = document.createElement('input');
+    var id = 'lux_' + goog.getUid(this) + '_' + goog.getUid(layer);
+    input.type = 'checkbox';
+    input.id = id;
+    input.checked = layer.get('visible');
+    input.onchange = function(e) {
+      layer.set('visible', e.target.checked);
+    };
+    li.appendChild(input);
```
Comment (tonio) [inline jsapi/src/layermanager.js]: Maybe better wrap input into label to avoid id generation & simplify this a bit?

**Pull request merged**